src/nrtk_explorer/app/core.py

```diff
@@ -162,14 +158,11 @@ def on_random_sampling_change(self, **kwargs):
         self.reload_images()
 
     def reload_images(self):
-        with open(self.state.current_dataset) as f:
-            dataset = json.load(f)
-
         categories = {}
-        for category in dataset["categories"]:
+        for category in self.context.dataset.cats.values():
```

Comment:
`self.context.dataset.dataset['categories']` for similar reasons to other comments.